ios/RNOneginiSdk.swift

```diff
@@ -176,21 +182,31 @@ class RNOneginiSdk: RCTEventEmitter, ConnectorToRNBridgeProtocol {
             reject(String(WrapperError.malformedUrl.code), WrapperError.malformedUrl.description, WrapperError.malformedUrl)
             return
         }
-
-        if (bridgeConnector.toRegistrationConnector.registrationHandler.processRedirectURL(urlOBject)) {
-            resolve(nil)
-        } else {
-            reject(String(WrapperError.registrationNotInProgress.code), WrapperError.registrationNotInProgress.description, WrapperError.registrationNotInProgress)
+        do {
+            try bridgeConnector.toRegistrationConnector.registrationHandler.processRedirectURL(urlOBject)
+            return resolve(nil)
+        } catch {
+            reject(String(error.code), error.localizedDescription, error)
         }
     }
 
     @objc
     func cancelRegistration(_ resolve: @escaping RCTPromiseResolveBlock,
                             rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
         let registrationHandler = bridgeConnector.toRegistrationConnector.registrationHandler
-        var canceled = registrationHandler.cancelBrowserRegistration()
-        canceled = registrationHandler.cancelPinCreation() || canceled
-        canceled = registrationHandler.cancelCustomRegistration() || canceled
+        var canceled = false
+        do {
+            try registrationHandler.cancelBrowserRegistration()
+            canceled = true
+        } catch {}
```

Comment:
If you are not interested in error you can have only
```
try? registrationHandler.cancelBrowserRegistration()
canceled = true

Reply:
I rewrote this to something else, but I'm not convinced about the structure.

Reply:
@szymon-onewelcome lets not focus too much on this, as this is getting a rework in RNP-126. This cancelRegistration api is actually getting removed.

Reply:
Reworked here https://github.com/onewelcome/sdk-react-native/pull/96